tests/sentry/event_manager/grouping/test_assign_to_group.py

```diff
@@ -0,0 +1,408 @@
+            wraps=wrapped_calculate_secondary_hash,
+        ) as calculate_secondary_hash_spy,
+        mock.patch(
+            "sentry.event_manager._create_group",
+            wraps=_create_group,
+        ) as create_group_spy,
+    ):
+        yield {
+            "find_existing_grouphash": find_existing_grouphash_spy,
+            "_calculate_primary_hash": calculate_primary_hash_spy,
+            "_calculate_secondary_hash": calculate_secondary_hash_spy,
+            "_create_group": create_group_spy,
+        }
+
+
+def set_grouping_configs(
+    project: Project,
+    primary_config: str,
+    secondary_config: str | None,
+    transition_expiry: float | None = None,
+    in_transition: bool = False,
+):
+    project.update_option("sentry:grouping_config", primary_config)
+    project.update_option("sentry:secondary_grouping_config", secondary_config)
+    if in_transition:
+        project.update_option(
+            "sentry:secondary_grouping_expiry", transition_expiry or time() + 3600
+        )
+    else:
+        project.update_option("sentry:secondary_grouping_expiry", None)
+
+
+def save_event_with_grouping_config(
+    event_data: dict[str, Any],
+    project: Project,
+    primary_config: str,
+    secondary_config: str | None = None,
+    in_transition: bool = False,
+) -> Event:
+    """
+    Create an event with the given grouping config, by temporarily changing project options before
+    saving an event. Resets options to current values once the event is saved.
+    """
+    current_primary_config = project.get_option("sentry:grouping_config")
+    current_secondary_config = project.get_option("sentry:secondary_grouping_config")
+    current_transition_expiry = project.get_option("sentry:secondary_grouping_expiry")
+
+    set_grouping_configs(
+        project=project,
+        primary_config=primary_config,
+        secondary_config=secondary_config,
+        in_transition=in_transition,
+    )
+    event = save_new_event(event_data, project)
+
+    # Reset project options
+    set_grouping_configs(
+        project=project,
+        primary_config=current_primary_config,
+        secondary_config=current_secondary_config,
+        transition_expiry=current_transition_expiry,
+        in_transition=True,  # Force transition expiry to be set, even if it's None
+    )
+
+    return event
+
+
+def get_results_from_saving_event(
+    event_data: dict[str, Any],
+    project: Project,
+    primary_config: str,
+    secondary_config: str,
+    in_transition: bool,
+    existing_group_id: int | None = None,
+):
+    # Slightly gross (and with even grosser auto-formatting), but it saves having to do a bunch of
+    # `x if y else None` below
+    secondary_hash_calculated = (
+        hashes_different
+    ) = (
+        secondary_hash_found
+    ) = (
+        event_assigned_to_given_existing_group
+    ) = secondary_grouphash_existed_already = secondary_grouphash_exists_now = None
```

Comment:
I see what you're trying here, but with this auto-formatting it's actually longer and less readable than naive version

```suggestion
    secondary_hash_calculated = None
    hashes_different = None
    secondary_hash_found = None
    event_assigned_to_given_existing_group = None
    secondary_grouphash_existed_already = None
    secondary_grouphash_exists_now = None
```

Reply:
Yeah, that's probably true. I'm going to need to push again once the PR under this one merges, so I'll make the change then.